src/redisraft.c

```diff
@@ -509,6 +509,19 @@ static void handleMigrateCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, RaftRedi
     raft_entry_release(entry);
 }
 
+static void  handleAsking(RedisRaftCtx *rr, RedisModuleCtx *ctx)
+{
+    Node *ret;
+    if (checkLeader(rr, ctx, &ret) == RR_ERROR) {
+        replyRedirect(ctx, -1, &ret->addr);
+    }
```

Comment:
```suggestion
    if (checkLeader(rr, ctx, NULL) == RR_ERROR) {
        return;
    }
```

This should be this way, right? `checkLeader()` already replies with "redirect".

Reply:
actually, not, when passed with &ret, it doesn't redirect (i.e. for proxy mode), but need to remember what I was doing here, as you're right, if redirecting, no need for that.  I think think the point was that this in an ASKING handling, and hence we might want to reply with an ask response and hence would need the leader.  In our case, we don't right now (ask doesn't work for us right now, as the ASK doesn't include the slot we are asking about and hence can't return a well formed ask response, as an aside, part of me thinks that's also something missing from the protocol, an "ASK" response includes the slot its about, an ASKING should include it (and only work on said slot)